utils/api.go

```diff
@@ -31,18 +33,21 @@ func OriginChecker(allowedOrigins string) func(*http.Request) bool {
 }
 
 func RenderWebError(err *model.AppError, w http.ResponseWriter, r *http.Request) {
-	message := err.Message
-	details := err.DetailedError
-
 	status := http.StatusTemporaryRedirect
 	if err.StatusCode != http.StatusInternalServerError {
 		status = err.StatusCode
 	}
 
-	http.Redirect(
-		w,
-		r,
-		"/error?message="+url.QueryEscape(message)+
-			"&details="+url.QueryEscape(details),
-		status)
+	destination := strings.TrimRight(GetSiteURL(), "/") + "/error?message=" + url.QueryEscape(err.Message)
+	if status >= 300 && status < 400 {
+		http.Redirect(w, r, destination, status)
+		return
+	}
+
+	w.WriteHeader(status)
+	fmt.Fprintln(w, `<!DOCTYPE html><html><head></head>`)
```

Comment:
Should we use an HTML template for this?

Also, to make sure I'm understanding correctly, we're using the meta tags to ensure that the user is properly redirected to the error page in case their browser doesn't support redirects on a 4xx or 5xx error code?

Reply:
0/5 on using a template. This just seemed a little simpler to me.

This doesn't attempt to do an HTTP redirect at all for 400+ error codes. It's not valid HTTP and I'm not aware of *any* browsers that handle it properly in all cases. For example, Chrome with a 501 status code ignores the redirect, so the user previously just saw an unstyled page that said "Not Supported" instead of our custom error page.

This attempts a JavaScript redirect. If JavaScript is disabled, it attempts a meta-refresh redirect. If for some reason that doesn't work, there's a subtle link that the user can click (but this is not the norm and will likely never happen).

Reply:
I'm fine with inlining it instead of using a template. I was going to ask you to change it for consistency, but then I realized that this is an edge case of an error case, so why waste the effort on that